examples/vue3/src/generateSignatureIfSecret.js

```diff
@@ -0,0 +1,34 @@
+const enc = new TextEncoder('utf-8')
+async function sign (secret, body) {
+  const algorithm = { name: 'HMAC', hash: 'SHA-384' }
+
+  const key = await crypto.subtle.importKey('raw', enc.encode(secret), algorithm, false, ['sign', 'verify'])
+  const signature = await crypto.subtle.sign(algorithm.name, key, enc.encode(body))
+  return `sha384:${Array.from(new Uint8Array(signature), x => x.toString(16).padStart(2, '0')).join('')}`
+}
+function getExpiration (future) {
+  return new Date(Date.now() + future)
+    .toISOString()
+    .replace('T', ' ')
+    .replace(/\.\d+Z$/, '+00:00')
+}
+/**
+ * Adds an expiration date and signs the params object if a secret is passed to
+ * it. If no secret is given, it returns the same object.
+ *
+ * @param {string | undefined} secret
+ * @param {object} params
+ * @returns {{ params: string, signature?: string }}
+ */
+export default async function generateSignatureIfSecret (secret, params) {
```

Comment:
Didn't we have an ongoing discussion whether we should recommend the secret to be exposed or not? Might be good to conclude that before making it the official example.

Reply:
I've added a comment about that: https://github.com/transloadit/uppy/pull/3774/files#diff-c7f113e07543dc23344eec96338470c18b3e53c0fa8c0dfe01d0d49c8ab76c9fR71-R72
There's not much we can do until we have an other solution